docs/design/resource-states.md

```diff
@@ -0,0 +1,222 @@
+`Reason` is derived from multiple sources depending on where in the resource lifecycle we are. The value in `Reason` may be set by the operator itself, or be set to the result of an error or response code received from Azure. 
+
+The operator defines the following reasons:
+
+| Reason                 | Severity                        |   Meaning                                                                                                                                                                 |
+|------------------------|---------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Reconciling            | Info                            | A request has been submitted to Azure. No error has been received from Azure, but one may occur in the future.                                                            |
+| WaitingForOwner        | Warning                         | The `owner` of this resource cannot be found in Kubernetes. Progress is blocked until the `owner` is created.                                                             |
+| Deleting               | Info                            | The resource is being deleted.                                                                                                                                            |
+| Succeeded              | None (""/empty)                 | The `spec` has successfully been applied. No additional changes are being attempted at this time.                                                                         |
+
+There are no failure conditions (`Severity = Error`) specified here because currently all fatal errors come directly from Azure. When an error response is received from Azure, the `Code` from Azure is set as the `Reason`, and the `Message` from Azure
+is set as the `Message`.
+
+#### `Severity` meaning in the context of the `Ready` condition
+
+* `Info` means that everything is proceeded as expected. This is the "happy path".
+* `Warning` means that something is wrong, but we haven't given up. The resource will continue to be reconciled until we can progress past the cause of the `Warning`. Users should examine the `Warning` as some may be due to the operator waiting for action from them, as in the case of `Reason=WaitingForOwner, Severity=Warning`. Others may be due to transient unavailability in Azure, as in the case of `Reason=InternalServerError, Severity=Warning`.
+* `Error` means that we were unable to reconcile the resource successfully. A `Ready` condition with `Status=False, Reason=<something>, Severity=Error` is no longer attempting to be reconciled. The user must make an update to the resource to fix one or more configuration problems.
```

Comment:
Maybe list these in the reverse order, starting from `Error`?

Reply:
Done